src/librustc/dep_graph/dep_node.rs

```diff
@@ -331,7 +331,7 @@ macro_rules! define_dep_nodes {
             /// refers to something from the previous compilation session that
             /// has been removed.
             #[inline]
-            pub fn extract_def_id(&self, tcx: TyCtxt) -> Option<DefId> {
+            pub fn extract_def_id(&self, tcx: TyCtxt<'a, 'gcx, 'tcx>) -> Option<DefId> {
```

Comment:
I think I would prefer this as '_ since they're not used.

Reply:
I too would prefer to use `'_` — I like to know when the names are important. I also just don't like our convention of using name `'a`, though I guess that's neither here nor there =)

Reply:
Are we using in-band lifetimes here, or are these names in scope?